lang/cs/Org.Apache.REEF.IMRU/OnREEF/IMRUTasks/MapTaskHost.cs

```diff
@@ -59,15 +60,16 @@ private MapTaskHost(IMapFunction<TMapInput, TMapOutput> mapTask, IGroupCommClien
         /// <returns></returns>
         public byte[] Call(byte[] memento)
         {
-            MapInputWithControlMessage<TMapInput> mapInput = _dataAndMessageReceiver.Receive();
-
-            while (mapInput.ControlMessage == MapControlMessage.AnotherRound)
+            while (true)
             {
+                GC.Collect();
```

Comment:
yes when container reaches 8 gb it just gets killed which means Gc uses
overall computer memory for calculations.\ on when to get invoked.

On Tue, Sep 29, 2015 at 8:47 AM, Markus Weimer <notifications@github.com>
wrote:

> In lang/cs/Org.Apache.REEF.IMRU/OnREEF/IMRUTasks/MapTaskHost.cs
> <https://github.com/apache/incubator-reef/pull/527#discussion_r40689383>:
>
> >              {
> > +                GC.Collect();
>
> To give you example, with 50 million dimensional vectors and with 1.4 GB
> of data in memory per mapper, the peak memory usage of evaluator is 8GB
> without GC.Collect and 2.8 GB with GC.Collect.
>
> This indeed is significant, but only if we need to stay at the lower
> memory usage. Is the container in question 8GB? If so, the GC will be
> called automatically when we reach the limit. Which would incur less
> frequent execution pauses due to the GC.
>
> You might also experience another issue: Depending on how the cluster is
> configured, the CLR sees the available memory to the machine, not the
> available memory of the container. If you are on a 128GB machine and have a
> 4GB container, this can create a situation in which the CLR sees no reason
> to do garbage collection early enough.
>
> However, that is a cluster configuration issue, and none we should
> prematurely address in REEF.
>
> We can give user an option for this. i.e. in IMRU give a configurable
> parameters say Boolean called MemoryEfficient that allows user to trade off
> memory efficiency with overhead of GC.
>
> Yes, that is definitely better. However, I'd go for a more neutral name
> like InvokeGarbageCollectionAfterCommunications as, given the choice,
> most people would want to be MemoryEfficient, even if it is unclear what
> that means :-)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-reef/pull/527/files#r40689383>.
>